backend/btrixcloud/crawlconfigs.py

```diff
@@ -164,108 +164,97 @@ def sanitize(self, string=""):
+            config=config_in.config,
+            name=config_in.name,
+            description=config_in.description,
+            tags=config_in.tags,
+            jobType=config_in.jobType,
+            crawlTimeout=config_in.crawlTimeout,
+            maxCrawlSize=config_in.maxCrawlSize,
+            scale=config_in.scale,
+            autoAddCollections=config_in.autoAddCollections,
+            profileid=config_in.profileid,
+            crawlerChannel=config_in.crawlerChannel,
+            crawlFilenameTemplate=config_in.crawlFilenameTemplate,
         )
 
-        run_now = config.runNow
-        storage_quota_reached = await self.org_ops.storage_quota_reached(org.id)
-        exec_mins_quota_reached = await self.org_ops.exec_mins_quota_reached(org.id)
-
-        if org.readOnly:
-            run_now = False
-            print(f"Org {org.id} set to read-only", flush=True)
+        if config_in.runNow:
+            crawlconfig.lastStartedBy = user.id
+            crawlconfig.lastStartedByName = user.name
 
-        if storage_quota_reached:
-            run_now = False
-            print(f"Storage quota exceeded for org {org.id}", flush=True)
+        # Ensure page limit is below org maxPagesPerCall if set
+        max_pages = await self.org_ops.get_max_pages_per_crawl(org.id)
+        if max_pages > 0:
+            crawlconfig.config.limit = max_pages
 
-        if exec_mins_quota_reached:
-            run_now = False
-            print(f"Execution minutes quota exceeded for org {org.id}", flush=True)
+        # add  CrawlConfig to DB here
+        result = await self.crawl_configs.insert_one(crawlconfig.to_dict())
 
         await self.crawl_manager.update_scheduled_job(crawlconfig, str(user.id))
 
         crawl_id = None
+        storage_quota_reached = False
+        exec_mins_quota_reached = False
 
-        if run_now:
-            crawl_id = await self.crawl_manager.create_crawl_job(
-                crawlconfig,
-                org.storage,
-                userid=str(crawlconfig.modifiedBy),
-                warc_prefix=self.get_warc_prefix(org, crawlconfig),
-                storage_filename=storage_filename,
-                profile_filename=profile_filename or "",
-            )
-
-            await self.add_new_crawl(crawl_id, crawlconfig, user, manual=True)
-
-        return (
-            result.inserted_id,
-            crawl_id,
-            storage_quota_reached,
-            exec_mins_quota_reached,
+        if config_in.runNow:
+            try:
+                crawl_id = await self.run_now_internal(crawlconfig, org, user)
+            except HTTPException as e:
+                if e.detail == "storage_quota_reached":
+                    storage_quota_reached = True
+                elif e.detail == "exec_minutes_quota_reached":
+                    exec_mins_quota_reached = True
+                print(f"can't run now: {e.detail}")
+        else:
+            # just compute quotas anyway?
```

Comment:
Yes, this will help the frontend reset if necessary so good to include. Can remove this comment

```suggestion
```